### ISSUE TITLE
Request for a feature to validate import statements that import a default export

### ISSUE THREAD

**Issue by Zamiell** [benmosher/eslint-plugin-import#1535]
Greetings and thanks for the excellent plugin.

I have a request for a new feature that would be very helpful for catching some obvious buggy import statements!

Background:
* Today, I was converting my JavaScript project code-base from CommonJS-style imports to ES6-style imports, e.g. `const foo = require('foo');` --> `import * as foo from 'foo';`
* About half of the JavaScript files in my project have a default export, and the other half have multiple exports. So, for any given file, my import statements at the top look something like this:

```javascript
import * as module1 from 'module1'; // Because module1 has multiple exports
import module2 from 'module2'; // Because module2 has one export, a default export
import * as module3 from 'module3'; // Because module3 has multiple exports
import module4 from 'module4'; // Because module4 has one export, a default export
```

However, at one point, I made a typo in an import statement that had a default import, like so:

```javascript
import * as module2 from 'module2'; // module2 only has one export, a default export, oops!
module2() // Epic run-time fail!
```

This error snuck by both eslint and webpack - I was only able to catch the error at runtime.

Currently, the inverse of this problem (trying to use a default import statement with a module that actually exports multiple things) is **already caught** by eslint-plugin-import, e.g. 
```
Caution: `module4` also has a named export `foo`. Check if you meant to write `import {foo} from './module4'` instead.
```

This is extremely helpful! So it strikes me that also validating for the inverse probably would potentially be easy to add and also very helpful.

Thanks. = )

**Comment by ljharb** [comment]
I’m not sure it is easy - generally I’d say, never use the `import *` format; only explicitly import the things you want, which avoids this problem entirely. import * is for laziness or metaprogramming, and no other purpose.

However, there’s nothing inherently wrong with importing a module namespace object from a module with only a default export.

**Comment by Zamiell** [comment]
> However, there’s nothing inherently wrong with importing a module namespace object from a module with only a default export.

I can see the use case for doing an `import *` on a module that has **both** a default export and some other exports. However, what would be a use case for doing an `import *` on a module that **only** has a default import?

**Comment by ljharb** [comment]
The use case is the same as every other case of using import * - you want an object to reflect on. If you don’t want that, using import * is incorrect entirely, no matter how many exports there are.

**Comment by Zamiell** [comment]
I see. Okay, thank you for the explanation. I see now that that the Airbnb style guide also disallows "import *", so I will change my code and close this issue.
